paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/FileStoreSourceSplitReader.java

```diff
@@ -121,6 +129,13 @@ public void handleSplitsChanges(SplitsChange<FileStoreSourceSplit> splitsChange)
         }
 
         splits.addAll(splitsChange.splits());
+        DataSplit dataSplit = (DataSplit) splits.peek().split();
+        if (sourceReaderMetrics != null
+                && (currentSnapshotId == FileStoreSourceReaderMetrics.UNDEFINED
+                        || dataSplit.snapshotId() != currentSnapshotId)) {
+            currentSnapshotId = dataSplit.snapshotId();
```

Comment:
You're now using `DataFileMeta` creation time instead of snapshot creation time. No need to compare `snapshotId`.

Reply:
Yes, there is no need to check if snapshot changes.

Reply:
Then I think the time update could be recorded at fetching each new split, which could be more gauge values filled in.